MetaLamp/nft-marketplace/src/Plutus/Contracts/NftMarketplace/OffChain/User.hs

```diff
@@ -165,14 +167,95 @@ closeSale marketplace CloseSaleParams {..} = do
     let ipfsCidHash = sha2_256 cspIpfsCid
     nftStore <- marketplaceStore marketplace
     nftEntry <- maybe (throwError "NFT has not been created") pure $ AssocMap.lookup ipfsCidHash nftStore
-    nftLot <- maybe (throwError "NFT has not been put on sale") pure $ nftEntry ^. Core._nftSale
+    nftSale <- maybe (throwError "NFT has not been put on sale") pure $
+                  nftEntry ^. Core._nftLot ^? traverse . Core._lotLink . _Left
 
-    _ <- Sale.redeemLot $ Sale.fromTuple $ nftLot ^. Core._lotSale
+    _ <- Sale.redeemLot $ Sale.fromTuple nftSale
 
     let client = Core.marketplaceClient marketplace
-    void $ mapError' $ runStep client $ Core.CloseSaleRedeemer ipfsCidHash
+    void $ mapError' $ runStep client $ Core.RemoveLotRedeemer ipfsCidHash
 
-    logInfo @Haskell.String $ printf "Closed NFT lot sale %s" (Haskell.show nftLot)
+    logInfo @Haskell.String $ printf "Closed NFT sale %s" (Haskell.show nftSale)
+    pure ()
+
+data HoldAnAuctionParams =
```

Comment:
I think it'd be nice to add initial price option.

Reply:
added a todo for that